### PR TITLE
Added new request functionality

### LINKED ISSUE #2354: Airlock UI: New Request
Implement as per https://www.figma.com/file/Ncl8TJsFf2WKf3frrqTbtq/AzureTRE-OSS-UI?node-id=305%3A21271

Showing file information depends on #2267, which is currently blocked/low priority, so this can be omitted for now

### PR DESCRIPTION
**Pull request by jjgriff93**
# Resolves  #2354

## What is being addressed

Implemented create airlock request functionality

## How is this addressed

- Created new routable component for new requests
- Implemented steps for selecting request type and filling out associated details
- Validation for request submission
- API call and error handling

image

image

### CODE REVIEW

File: ui/app/src/components/shared/airlock/Airlock.tsx [hunk 1]
```diff
@@ -68,18 +72,26 @@ export const Airlock: React.FunctionComponent<AirlockProps> = (props: AirlockPro
             newCol.isSortedDescending = true;
           }
         });
+        console.log(columns);
```
Comment (damoodamoo) [inline ui/app/src/components/shared/airlock/Airlock.tsx:75]: `nit` - do we need this log still?

File: ui/app/src/components/shared/airlock/AirlockNewRequest.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+import { ExceptionLayout } from "../ExceptionLayout";
+
+interface AirlockNewRequestProps {
+  onCreateRequest: (request: AirlockRequest) => void;
+}
+
+const stackTokens: IStackTokens = { childrenGap: 20 };
+const { palette, fonts } = getTheme();
+
+const importPreviewGraphic: IDocumentCardPreviewProps = {
```
Comment (damoodamoo) [inline ui/app/src/components/shared/airlock/AirlockNewRequest.tsx:18]: sort of a `nit` - can we move all the boilerplate / layout code to the bottom, as we are less likely to care about it?

File: ui/app/src/components/shared/airlock/AirlockNewRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        setApiSubmitError(err);
+        setCreateError(true);
+      }
+      setCreating(false);
+    }
+  }, [apiCall, newRequest, props, workspaceCtx, requestValid]);
+
+  const dismissPanel = useCallback(() => navigate('../'), [navigate]);
+
+  let title;
```
Comment (damoodamoo) [inline ui/app/src/components/shared/airlock/AirlockNewRequest.tsx:115]: can we add type hints here?

File: ui/app/src/components/shared/airlock/AirlockNewRequest.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        value={newRequest.businessJustification}
+        onChange={onChangeBusinessJustification}
+        multiline
+        rows={10}
+        required
+      />
+    </Stack>;
+  }
+
+  const renderFooter = useCallback(() => {
```
Comment (damoodamoo) [inline ui/app/src/components/shared/airlock/AirlockNewRequest.tsx:167]: think any hooks should be defined at the top of the function, definitely before any conditional logic

**Review by damoodamoo — APPROVED** [review]
looks good! Few smaller things but happy for you to resolve/ignore/merge as you want

File: ui/app/src/components/shared/airlock/Airlock.tsx (hunk 1 above)
Comment (jjgriff93) [reply ui/app/src/components/shared/airlock/Airlock.tsx:75]: Good spot, removed

File: ui/app/src/components/shared/airlock/AirlockNewRequest.tsx (hunk 3 above)
Comment (jjgriff93) [reply ui/app/src/components/shared/airlock/AirlockNewRequest.tsx:115]: Yep

File: ui/app/src/components/shared/airlock/AirlockNewRequest.tsx (hunk 2 above)
Comment (jjgriff93) [reply ui/app/src/components/shared/airlock/AirlockNewRequest.tsx:18]: Good idea

**Pull request merged**